src/cascadia/TerminalSettingsModel/CascadiaSettings.cpp

```diff
@@ -334,6 +336,24 @@ void CascadiaSettings::_ValidateProfilesExist()
     }
 }
 
+// Method Description:
+// - Checks if the profiles contain invalid environment variable values
+//   profiles at all, we'll throw an error if there aren't any profiles.
+void CascadiaSettings::_ValidateProfileEnvironmentVariables()
+{
+    for (const auto& profile : _allProfiles)
+    {
+        try
+        {
+            profile.ValidateEvaluatedEnvironmentVariables();
+        }
+        catch (...)
+        {
+            _warnings.Append(Microsoft::Terminal::Settings::Model::SettingsLoadWarnings::InvalidProfileEnvironmentVariables);

```

Comment:
Will that catch e.g. std::bad_alloc and give a misleading message?

Reply:
Yes it would, updated.